### ISSUE TITLE
Scanning takes a lot of time

### ISSUE THREAD

**Issue by alepulver** [meganz/MEGAsync#596]
I'm using MEGAsync 4.5.1 on Windows 10 x64 and after setting up a folder with 80k files it takes about 20 min for "scanning...". My drive is an SSD and I'm surprised it takes so long, as Total Commander calculates the occupied space, number of files and directories in a few seconds.

It would also be useful to have a progress bar or some kind of indicator during this process.
I tried checking the logs but they don't look like plain text.

**Comment by zDEFz** [comment]
Keep your machine on and shutdown as little as possible, sleep mode preferred. 

Also refer to #596
Actually part of the answer stems from one of the replies I got from the MEGA support.

**Comment by asheroto** [comment]
Scanning does seem to take a bit.

Consider the application [Everything](https://www.voidtools.com/) - it scans my **entire** hard drive in less than 20 seconds.  Whereas megasync will take a few minutes to scan just 6 sync folders.

I suppose it is using multiple threads to achieve this. Might be something you guys look into using.

**Comment by zDEFz** [comment]
> Scanning does seem to take a bit.
> 
> Consider the application [Everything](https://www.voidtools.com/) - it scans my **entire** hard drive in less than 20 seconds. Whereas megasync will take a few minutes to scan just 6 sync folders.
> 
> I suppose it is using multiple threads to achieve this. Might be something you guys look into using.

You are wrong. Scanning takes long because it checks serverside for file revisions.

**Comment by mattw-mega** [comment]
Hi @alepulver , you may like to know we have a big update coming for MEGAsync which should be much faster starting up, especially when the sync is already up to date. You can read about it and perhaps help test it here: https://github.com/meganz/MEGAsync/issues/770 thanks

**Comment by DanielJoseDolz** [comment]
I have hundres of thousand of file and the scanning never stops. It never uploads.
It is a shame Mega do not warn this. "up to 8TB and XXXX files". If there is no solution I have to confront the manager and tell her we wasted money and the design schema will not work

**Comment by zDEFz** [comment]
> I have hundres of thousand of file and the scanning never stops. It never uploads. It is a shame Mega do not warn this. "up to 8TB and XXXX files". If there is no solution I have to confront the manager and tell her we wasted money and the design schema will not work

MEGA is not suitable for folders with many files / nesting. The support told me that one should have instead small amount of big files instead to fill the storage.

**Comment by DanielJoseDolz** [comment]
> MEGA is not suitable for folders with many files / nesting. The support told me that one should have instead small amount of big files instead to fill the storage. You just don't use it properly I guess. Store locally, and then package into large archive files.

This warning in the pricing/features section would be nice.
Still working on it. Apparently the "backup" option could do it, but here is no versioning (it is a must for us because we are doing this out of ransomware fear). So I am having mixed feelings here.

**Comment by mattw-mega** [comment]
Hi @DanielJoseDolz , @zDEFz ,  we have a much improved version of MEGAsync coming up which is fine with extremely large numbers of files and folders, and with nesting, especially once the files and folders are in sync.   In particular, on windows it uses a much better Win32 function to retrieve the directory listings.   On the first scan, it has to fingerprint the files too, which does take a while.  But on second and subsequent scans, provided all other attributes are unchanged, it does not re-fingerprint the files.    You can try it out at #770 if you like.   For an up to date sync, you should find it very quick to start up and scan the existing syncs for any changes, and be back to idle.   Especially if the OS has cached the directory information.  thanks

**Comment by mattw-mega** [comment]
Hi @DanielJoseDolz , about the versioning in Backups.   Actually backups are very similar to syncs, and versioning works the same way.   If you have versioning turned on in your account, Backups will make versions.    However the webclient does not show those under the Backup area.   If you stop the Backup and move the folder tree to your main account area, you will see the versions are there.    This came up internally recently, and I'm sure that will be fixed in due course.  thanks

**Comment by communitydriven** [comment]
Hello i am using Meganz and i use int mainly to sync textfiles. I work mostly to write .txt files on windows 10 and sync it between my devices and i love to have the text file editor on my phone app.  What i love is that i can show my textfile on the app and edit it when i click on edit. And it is darkmode. No other cloud has this. My textfiles are about 2 between 5kb (kilobytes) and sometimes i am facing that it gets stuck and i have to create a new textfile and copy the whole file into the new textfile that it can sync. I would be happy that it do not get stuck in the future. Thank you.

**Comment by biggestsonicfan** [comment]
Did anything ever happen with this? Been "scanning" for about six hours now. I paused a transfer and rebooted and I lost my entire queue somehow. So....

**Comment by rodrihalo117** [comment]
> Did anything ever happen with this? Been "scanning" for about six hours now. I paused a transfer and rebooted and I lost my entire queue somehow. So....

Same here, waited all the day and it's still "scanning"

**Comment by rautamiekka** [comment]
> Did anything ever happen with this? Been "scanning" for about six hours now. I paused a transfer and rebooted and I lost my entire queue somehow. So....

> > Did anything ever happen with this? Been "scanning" for about six hours now. I paused a transfer and rebooted and I lost my entire queue somehow. So....
> 
> Same here, waited all the day and it's still "scanning"

Did you keep an eye on the task manager ? Writing the SQLite databases & the pure-text log files ain't exactly fast, especially if the to-be-scanned files are on the same drive as your home (user profile) folder.

**Comment by biggestsonicfan** [comment]
I didn't. I just literally assumed those would have been propagated when it had them all in the queue. I am unsure why I had to restart from the beginning. I've now reached a point where a bunch of 0 byte files serve as placeholders and resuming works.

**Comment by LuisMCunha** [comment]
Same here, it's been scanning for multiple days now with the PC always on.
This for 26GB worth of files, it's kinda ridiculous.

**Comment by johentsch** [comment]
Same here, something must be wrong, definitely:
![image](https://github.com/meganz/MEGAsync/assets/42718519/fd123872-b8fb-4247-a459-545e2d282bcc)
It just keeps scanning new files. 37 million??? Both `ncdu` and `find .//. ! -name . -print | grep -c //` tell me the folder recursively includes 340600 files only! Seems like an infinite loop to me...

This is `megasync 2.5.1-5` running on Arch Linux.

**Comment by rautamiekka** [comment]
> Same here, something must be wrong, definitely: ![image](https://private-user-images.githubusercontent.com/42718519/345374571-fd123872-b8fb-4247-a459-545e2d282bcc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjAwMjYxMDUsIm5iZiI6MTcyMDAyNTgwNSwicGF0aCI6Ii80MjcxODUxOS8zNDUzNzQ1NzEtZmQxMjM4NzItYjhmYi00MjQ3LWE0NTktNTQ1ZTJkMjgyYmNjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzAzVDE2NTY0NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk1ZTAzMTIxNjVjYzVkY2JjMjg1OGI2Y2M1YjUzNGJkNThmZGQ0OTY5MDgxYjdjMjc0NGUxZTgyNWM5MWZlMDUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.esUiYc_h6EfPJ8OgyJjHNSwGELGbu9fqOQNTTLwMmns) It just keeps scanning new files. 37 million??? Both `ncdu` and `find .//. ! -name . -print | grep -c //` tell me the folder recursively includes 340600 files only! Seems like an infinite loop to me...
> 
> This is `megasync 2.5.1-5` running on Arch Linux.

That's ancient ! Hell, way older than OP's. You can literally sum the numbers together and it'd still be far less than the current newest, which has at least mostly fixed the problem (I haven't tried to sync a tree of **millions** of files since then, though).

**Comment by johentsch** [comment]
Hahaha, I'm sorry, I swapped the first digits, version is `5.2.1-5`.

**Comment by ACHREF999** [comment]
man this scanning is taking alot of time
any work arounds or smth

**Comment by Freeman999** [comment]
My main folder is quite big, Windows tells me: 117 GB, 119500 Files, 11300 Folders
As it has quite a few under-folders right one step below it, I dont really want do sync them all individually.
It syncs now since 10+ hours and it still reads 0 files, 1 folder.
<img width="630" height="22" alt="Image" src="https://github.com/user-attachments/assets/b47049de-1ffd-48df-82f7-4e70bd3aa361" />
Every few seconds it switches to "Ordnerscan" (scanning folders).
Any ideas?

**Comment by rl-mega** [comment]
Hi @Freeman999, if you haven't done it, please contact our support team through the "Report issue" ("Problem melden") button in the "General" tab of the Settings dialog. Please include logs (there is a pre-checked checkbox to that effect in the "report issue" dialog) so we can assess what is happening.
Thanks in advance and have a good day.

**Comment by rl-mega** [comment]
@ACHREF999 First scan can be very long depending on the number of files and folders. Subsequent scans and monitoring is much faster :)
